src/context.ts

```diff
@@ -33,10 +33,10 @@ export const MagicalContext = createContext<MagicalContext>({
 
 export const withWizardContext = <WrappedComponentProps = any>(
   Component: React.ComponentType<WrappedComponentProps>,
-) => {
-  const wizardProps = useContext(MagicalContext);
+): Function => {
```

Comment:
You should be more specific in typing here and write smth like React.FC<CombinedProps>, because `Function` doesn't have any constraints rather than it's a function, the return type will be Function, so this HOC then can be used in a way that's not supposed to be used, and TS will not throw error because type is not as strict as it needs to be

Reply:
✅